server/index.ts

```diff
@@ -47,125 +56,255 @@ app.get('/auth/login', (_req, res) => {
     scope: scope.join(' '),
     redirect_uri: spotifyRedirectUri,
     state: state,
-  } as URLProps);
+  } as LoginUrlProps);
 
   res.redirect(
     `https://accounts.spotify.com/authorize/?${authQueryParameters.toString()}`,
   );
 });
 
-// callback endpoint
-app.get('/auth/callback', (req, res) => {
+// callback endpoint -- set token server globals and redirect to homepage
+app.get('/auth/callback', async (req, res) => {
   const { code } = req.query;
+  const formData = new URLSearchParams({
+    code,
+    redirect_uri: spotifyRedirectUri,
+    grant_type: 'authorization_code',
+  } as CbUrlProps).toString();
 
-  const authOptions = {
-    url: 'https://accounts.spotify.com/api/token',
-    form: {
-      code: code,
-      redirect_uri: spotifyRedirectUri,
-      grant_type: 'authorization_code',
-    },
-    headers: {
-      Authorization: `Basic ${Buffer.from(
-        `${spotifyClientId}:${spotifyClientSecret}`,
-      ).toString('base64')}`,
-      'Content-Type': 'application/x-www-form-urlencoded',
-    },
-    json: true,
-  };
-
-  request.post(authOptions, function (error, response, body) {
-    if (!error && response.statusCode === 200) {
-      accessToken = body.access_token;
-      res.redirect('http://localhost:3000');
-    } else {
-      console.log(
-        error,
-        response.body.error,
-        response.body.error.error_description,
-      );
-      res.end(response.body);
-    }
-  });
+  try {
+    const response = await axios({
+      method: 'post',
+      url: 'https://accounts.spotify.com/api/token',
+      data: formData,
+      headers: {
+        Authorization: `Basic ${Buffer.from(
+          `${spotifyClientId}:${spotifyClientSecret}`,
+        ).toString('base64')}`,
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    });
+    accessToken = response.data.access_token;
+    expiresIn = response.data.expires_in;
+    refreshToken = response.data.refresh_token;
+    res.redirect('http://localhost:3000');
+  } catch (err) {
+    console.log(err);
+    res.send(err);
+  }
 });
 
-// token endpoint
+// token endpoint -- send server globals to front-end
 app.get('/auth/token', (_req, res) => {
-  res.json({ access_token: accessToken });
+  const tokenObj = {
+    access_token: accessToken,
+    expires_in: expiresIn,
+    refresh_token: refreshToken,
+  };
+  res.json(tokenObj);
 });
 
-// userInfo endpoint
-app.get('/auth/me', (_req, res) => {
-  const authMeOptions = {
-    url: 'https://api.spotify.com/v1/me',
-    headers: {
-      Authorization: `Bearer ${accessToken}`,
-      'Content-Type': 'application/json',
-    },
+// refresh token endpoint -- call endpoint to refresh server globals and send to front-end
+app.get('/auth/refresh_token/:rTok', async (req, res) => {
```

Comment:
In situations like this where you're only using the `req` object in one or maybe two ways, destructure it:
```suggestion
app.get('/auth/refresh_token/:rTok', async ({ params }, res) => {
```